add reference

Danielsilveira98 opened the pull request:
## Reference
Será necessário enviar o reference do usuário no data do push para que a SDK Kotlin consiga enviar o evento de receive, mesmo que o usuário não esteja identificado no aparelho. Anteriormente, quando o usuário não estava identificado, o evento era salvo com o id fazio, e isso causava erro no momento de envio para a Dito. Com a alteração vigente, e uma alteração na SDK, ao receber o push o evento de receive é enviado, ou salvo, com o id presente no referece do push.

Danielsilveira98 committed:
add reference

brenoassp committed:
fix(reference): BOPE-600 - altera de onde é pego os dados da reference, id e message da notificação

Modificação para passar a pegar os dados da notificação pelo data. Além disso, foi adicionado no data a reference da notificação. Dessa forma, independente se o usuário tiver logado ou não, a reference será passada para frente quando o usuário clicar na notificação recebida.

brenoassp committed:
fix(reference): BOPE-598 - remove else desnecessário

Diley commented on DitoSDK/sdk/src/main/java/br/com/dito/ditosdk/tracking/Tracker.kt:
```diff
@@ -107,11 +107,11 @@ internal class Tracker(private var apiKey: String, apiSecret: String, private va
             val data = JsonObject()
-            data.addProperty("identifier", id)
-            data.addProperty("reference", reference)
+            data.addProperty("identifier", notificationReference.substring(5))
```
@Danielsilveira98  O que esse substring faz?

Danielsilveira98 replied:
Ele pega da 5 posição da string para frente. Funciona como `reference[5..-1]` no ruby. Como o que vem da notificação é o reference do usuário, precisamos subtrair o id dele do reference.

Diley commented on DitoSDK/sdk/src/main/java/br/com/dito/ditosdk/notification/NotificationOpenedReceiver.kt:
```diff
@@ -17,23 +17,29 @@ class NotificationOpenedReceiver: BroadcastReceiver() {
         context?.let {
             getIntent(it)?.let {
-                context.startActivity(intent)
+                context.startActivity(it)
```
@Danielsilveira98 Pq mudamos isso?

Danielsilveira98 replied:
`getIntent` retorna o intent, e o `.let` faz o `intent` retornado ser jogado para a variável `it` . Da forma como estava antes, não estava encontrando o `intent`. Então alteramos que que o `intent` ativiado seja o que foi retornado pelo `getIntent`.

brenoassp committed:
fix exception that was not being handled when there was no reference

brenoassp committed:
fix: add FLAG_ACTIVITY_NEW_TASK to intent to avoid app crash

without this flag, if the user specify an option with an intent in the sdk initialization, the app crashes

brenoassp committed:
add new activity to test deep link

brenoassp committed:
fix(deep-link): change to open activity using deep link if it is present in the notification

brenoassp committed:
fix(deep-link): change code to not crash when the user sends an invalid deep link

brenoassp committed:
fix(deeplink): change to allow deeplink work if there is no reference in notification

brenoassp committed:
fix: change to use null as empty value instead of an empty string

brenoassp committed:
fix(deeplink): change to pass deeplink as an extra in the intent

also, add flag PendingIntent.FLAG_UPDATE_CURRENT to be able to update intent extras in future function calls
